src/dash-adapter.js

```diff
@@ -212,6 +213,11 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
   _init(): void {
     //Need to call this again cause we are uninstalling the VTTCue polyfill to avoid collisions with other libs
     shaka.polyfill.installAll();
+    if (this._config.useJsonp){
+      this._callback =  this._config.callback;
+      shaka.net.NetworkingEngine.registerScheme('http', HttpCorsPlugin.bind(this._callback), shaka.net.NetworkingEngine.PluginPriority.APPLICATION);
```

Comment:
Need to check if `registerRequestFilter` is more suitable for us, not sure it's promise based.